extensions/Worker.Extensions.Http.AspNetCore/src/HttpUriProvider.cs

```diff
@@ -21,7 +21,9 @@ public static Uri GetHttpUri()
             // TODO: replace local host string
             var uriString = "http://localhost:" + HttpPort.Value.ToString();
 
-            return new Uri(uriString);
+            _httpUri = new Uri(uriString);
```

Comment:
bug, unrelated to IActionResult but value wasn't being cached

Reply:
I noticed this as well. But another idea is to use a Lazy<string> instead of Lazy<int> and to the entire calculation and ToString() there -- then the value is cached in the Lazy instead of having to call ToString() multiple times.